src/react/tests/useMutation.test.tsx

```diff
@@ -519,4 +519,56 @@ describe('useMutation', () => {
 
     consoleMock.mockRestore()
   })
+
+  it('should pass meta to mutation', async () => {
+    const successMock = jest.fn()
+    const errorMock = jest.fn()
+
+    const queryClientAy = new QueryClient({
+      mutationCache: new MutationCache({
+        onSuccess: (_, __, ___, mutation) => {
+          successMock(mutation.meta?.mySuccessMessage)
+        },
+        onError: (_, __, ___, mutation) => {
+          errorMock(mutation.meta?.myErrorMessage)
+        },
+      }),
+    })
+
+    const mySuccessMessage = 'mutation succeeded'
+    const myErrorMessage = 'mutation succeeded'
+
+    function Page() {
+      const { mutate: succeed } = useMutation(async () => '', {
+        meta: { mySuccessMessage },
+      })
+      const { mutate: error } = useMutation(
+        async () => {
+          throw new Error('')
+        },
+        {
+          meta: { myErrorMessage },
+        }
+      )
+
+      return (
+        <div>
+          <button onClick={() => succeed()}>succeed</button>
+          <button onClick={() => error()}>error now</button>
+        </div>
+      )
+    }
+
+    const { getByText } = renderWithClient(queryClientAy, <Page />)
+
+    fireEvent.click(getByText('succeed'))
+    fireEvent.click(getByText('error now'))
+
+    await sleep(500)
```

Comment:
instead of waiting an arbitrary time, we should probably just render out the error returned from the mutation and wait for it to appear on the screen (the test right above does something similar)

Reply:
is that how it should be? The test checks if the 'global' `onSuccess` and `onError` are called. They shouldn't change something of the render of a component.

Reply:
5ea0c1c

Reply:
that solution also works, but is not really what I meant. I think he asserting that the callbacks have been called were totally fine.

to get around the sleep, I would've just rendered any error message, like:

```
{ isError ? <h1>Oh noes</div> : null }
```

and then instead of waiting 500ms, wait until that error is visible:

```
await waitFor(() => {
  expect(getByRole('heading').textContent).toBe('Oh noes')
})
```

then you have basically waited until the error mutation is finished, in which case both callbacks should've been called.

Reply:
but then where is the actual test that `meta` has been passed?

Reply:
like you had it before - with `expect(errorMock).toHaveBeenCalledWith(myErrorMessage)` ?

Reply:
4699132